packages/plugin-pnpm/sources/PnpmLinker.ts

```diff
@@ -0,0 +1,275 @@
+      return;
+
+    // We don't install those packages at all, because they can't be used anyway
+    if (!isPnpmVirtualCompatible(locator, {project: this.opts.project}))
+      return;
+
+    const pkgPath = this.locations.get(locator.locatorHash);
+    if (typeof pkgPath === `undefined`)
+      throw new Error(`Assertion failed: Expected the package to have been registered (${structUtils.stringifyLocator(locator)})`);
+
+    const nmPath = ppath.join(pkgPath, Filename.nodeModules);
+    if (dependencies.length > 0)
+      await xfs.mkdirpPromise(nmPath);
+
+    // Retrieve what's currently inside the package's true nm folder. We
+    // will use that to figure out what are the extraneous entries we'll
+    // need to remove.
+    const extraneous = await getNodeModulesListing(nmPath);
+
+    for (const [descriptor, dependency] of dependencies) {
+      // Downgrade virtual workspaces (cf isPnpmVirtualCompatible's documentation)
+      let targetDependency = dependency;
+      if (!isPnpmVirtualCompatible(dependency, {project: this.opts.project})) {
+        this.opts.report.reportWarning(MessageName.UNNAMED, `The pnpm linker doesn't support providing different versions to workspaces' peer dependencies`);
+        targetDependency = structUtils.devirtualizeLocator(dependency);
+      }
+
+      const depSrcPath = this.locations.get(targetDependency.locatorHash);
+      if (typeof depSrcPath === `undefined`)
+        throw new Error(`Assertion failed: Expected the package to have been registered (${structUtils.stringifyLocator(dependency)})`);
+
+      const name = structUtils.stringifyIdent(descriptor) as PortablePath;
+      const depDstPath = ppath.join(nmPath, name);
+
+      const depLinkPath = ppath.relative(ppath.dirname(depDstPath), depSrcPath);
+
+      const existing = extraneous.get(name);
+      extraneous.delete(name);
+
+      this.pendingPromises.push(Promise.resolve().then(async () => {
+        // No need to update the symlink if it's already the correct one
+        if (existing) {
+          if (existing.isSymbolicLink() && await xfs.readlinkPromise(depDstPath) === depLinkPath) {
+            return;
+          } else {
+            await xfs.removePromise(depDstPath);
+          }
+        }
+
+        await xfs.mkdirpPromise(ppath.dirname(depDstPath));
+
+        if (process.platform == `win32`) {
+          await xfs.symlinkPromise(depSrcPath, depDstPath, `junction`);
+        } else {
+          await xfs.symlinkPromise(depLinkPath, depDstPath);
+        }
+      }));
+    }
+
+    for (const name of extraneous.keys()) {
+      this.pendingPromises.push(xfs.removePromise(ppath.join(nmPath, name)));
+    }
+  }
+
+  async attachExternalDependents(locator: Locator, dependentPaths: Array<PortablePath>) {
+    throw new Error(`External dependencies haven't been implemented for the pnpm linker`);
+  }
+
+  async finalizeInstall() {
+    await Promise.all(this.pendingPromises);
```

Comment:
What about `.store` cleanup? Is it in the plans? When you remove dependency from the project, the corresponding entry in the `.store` will not be removed.

Reply:
Hmm that's a good point, I just forgot to implement some cleanup but yeah it should be made automatic 🤔

Reply:
Fixed